### ISSUE TITLE
build succeeds but loads with errors in browser (both dev and prod)

### ISSUE THREAD

**Issue by prabhatsharma** [mgechev/angular-seed#1583]
I did following:

git clone https://github.com/mgechev/angular-seed.git
cd angular-seed
npm install
npm run build.dev (This succeeds without any errors)
cd dist
cd dev
live-server --entry-file=index.html

I get following errors in browser console.

shim.min.js?1478174500884:1 Uncaught SyntaxError: Unexpected token <
Intl.min.js?1478174500885:1 Uncaught SyntaxError: Unexpected token <
system.src.js?1478174500885:1 Uncaught SyntaxError: Unexpected token <
system-config.ts:3 Uncaught ReferenceError: System is not defined(…)(anonymous function) @ system-config.ts:3
zone.js?1478174500887:1 Uncaught SyntaxError: Unexpected token <
Rx.min.js?1478174500887:1 Uncaught SyntaxError: Unexpected token <
(index):43 Uncaught ReferenceError: System is not defined(…)

$ node -v
v6.6.0

$ npm -v
3.10.8

$ uname -a
Linux blockmachine 4.4.0-45-generic #66-Ubuntu SMP Wed Oct 19 14:12:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

Browser - Chrome - Version 54.0.2840.59 unknown (64-bit)

angular2-seed commit hash - 3a13db9

**Comment by prabhatsharma** [comment]
I tried 
npm run build.prod.exp

Builds are working fine with this one. Was I doing something wrong in the initial command?

**Comment by mgechev** [comment]
That is not the way the project is intended to be used. As the readme file states you can start dev server & build with:

    npm start

For production use:

    npm run serve.prod

**Comment by prabhatsharma** [comment]
README says that we can get a prod build using npm run build.prod. Once we get the build files we can deploy them to any web server which in this case happened to be live-server (i was testing it). In fact I deployed the build to AWS S3 as well and I faced the same issue. Am I still missing something?

**Comment by TheDonDope** [comment]
The topic that needs configuration here is URL rewriting. The prod builds artifacts are the `index.html` as well as the bundled `app|shims.js` and `main.css`. Therefore, all URL requests need to go through the `index.html`, that is when the angular router kicks in and handles the navigation etc. I dont know how to do it this with `live-server`. What kind of webserver are you using in your AWS instance? As @mgechev said, you could simple use `$ npm run serve.prod`.
 
TL;DR: You need to configure your webserver to handle all requests to any route through the `index.html`

**Comment by prabhatsharma** [comment]
Thanks for replying. I do understand the part of URL rewriting and that is what 
live-server --entry-file=index.html

--entry-file parameter does. BTW I think I goofed up a little. prod build is working but not prod.exp (AOT) . Also I have hosted the build on Amazon S3 and set both index and error documents as index.html and it works pretty fine. I guess I am good for now for prod build. Dev build and AOT build is still not good. Running locally with npm works very fine and was not the problem that I was into and trying to solve. BTW, thanks for looking into it. http://b1.dev1.s3-website-us-east-1.amazonaws.com This URL is working pretty fine on S3. Thanks for making an awesome angular2 seed. My heartiest thanks to you guys.

**Comment by syedsuhail** [comment]
@prabhatsharma I am facing an issue with prod build in S3 as the injected app.js and shim.js are pointing to the default url. how did you change the inject url for these files ?

**Comment by prabhatsharma** [comment]
I have since then moved to angular-cli . Systemjs was becoming pain.
